### PR TITLE
Minor fixes for local execution

### PR DESCRIPTION
**Pull request by sundargates**
### Context

Explain context and other details for this pull request.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable

### CODE REVIEW

File: mantis-server/mantis-server-agent/src/main/resources/agent.properties [hunk 1]
```diff
@@ -18,8 +18,8 @@ mantis.taskexecutor.id=agent1
 mantis.localmode=true
 mantis.taskexecutor.metrics.collector=io.mantisrx.server.agent.metrics.cgroups.CgroupsMetricsCollector
 mantis.taskexecutor.rpc.port-range=5050
-mantis.taskexecutor.blob-store.storage-dir=file:///localstore_dir
-mantis.taskexecutor.blob-store.local-cache=/apps/mantis/mantis-server-agent/mantis-artifacts
+mantis.taskexecutor.blob-store.storage-dir=file:///Users/sananthanarayanan@netflix.com/Downloads/
+mantis.taskexecutor.blob-store.local-cache=/tmp/mantis/mantis-artifacts
```
Comment (sundargates) [inline mantis-server/mantis-server-agent/src/main/resources/agent.properties:22]: Rewrite this.

File: mantis-server/mantis-server-agent/build.gradle [hunk 2]
```diff
@@ -40,5 +40,5 @@ dependencies {
 mainClassName = "io.mantisrx.server.agent.AgentV2Main"
 
 run {
-    standardInput = System.in
+//    standardInput = System.in
 }
```
Comment (sundargates) [inline mantis-server/mantis-server-agent/build.gradle:44]: Get rid of this block completely

File: mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/MantisMasterClientApi.java [hunk 3]
```diff
@@ -668,6 +668,7 @@ private HttpClient<ByteBuf, ServerSentEvent> getRxnettySseClient(String hostname
 
     private WebSocketClient<TextWebSocketFrame, TextWebSocketFrame> getRxnettyWebSocketClient(String host,
                                                                                               int port, String uri) {
+        logger.info("Creating websocket client for " + host + ":" + port + " uri " + uri + " ...");
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/MantisMasterClientApi.java:671]: Get rid of this change

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/InMemoryPersistenceProvider.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.server.master.persistence;
+
+import io.mantisrx.master.events.LifecycleEventPublisher;
+import io.mantisrx.server.master.store.KeyValueStore;
+
+public class InMemoryPersistenceProvider extends KeyValueBasedPersistenceProvider {
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/InMemoryPersistenceProvider.java:22]: Move this to tests.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/KeyValueBasedPersistenceProvider.java [hunk 5]
```diff
@@ -756,4 +760,86 @@ public void addNewJobArtifact(JobArtifact jobArtifact) throws IOException {
...
+    @Override
+    public ResourceClusterScaleRulesWritable registerResourceClusterScaleRule(
+        ResourceClusterScaleRulesWritable ruleSpec) throws IOException {
+        return null;
+    }
+
+    @Override
+    public ResourceClusterScaleRulesWritable registerResourceClusterScaleRule(
+        ResourceClusterScaleSpec rule) throws IOException {
+        return null;
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/KeyValueBasedPersistenceProvider.java:838]: These need to be filled

File: mantis-control-plane/mantis-control-plane-server/src/test/java/io/mantisrx/master/api/akka/route/v1/ResourceClusterNonLeaderRedirectRouteTest.java [hunk 6]
```diff
@@ -89,15 +90,16 @@ public class ResourceClusterNonLeaderRedirectRouteTest extends JUnitRouteTest {
     private final ActorSystem system =
         ActorSystem.create(ResourceClusterNonLeaderRedirectRouteTest.class.getSimpleName());
 
+    private final IMantisPersistenceProvider storageProvider = new FileBasedPersistenceProvider(true);
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/test/java/io/mantisrx/master/api/akka/route/v1/ResourceClusterNonLeaderRedirectRouteTest.java:93]: Change this to InMemoryPersistenceProvider

File: mantis-control-plane/mantis-control-plane-server/src/test/java/io/mantisrx/master/resourcecluster/ResourceClusterAwareSchedulerActorTest.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import static org.mockito.Mockito.mock;
+
+import akka.actor.ActorSystem;
+import io.mantisrx.server.master.resourcecluster.ResourceCluster;
+import io.mantisrx.server.master.scheduler.ResourceClusterAwareScheduler;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ResourceClusterAwareSchedulerActorTest {
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/test/java/io/mantisrx/master/resourcecluster/ResourceClusterAwareSchedulerActorTest.java:28]: Delete this file

File: mantis-control-plane/mantis-control-plane-server/src/test/java/io/mantisrx/master/resourcecluster/ResourceClustersHostManagerActorTests.java [hunk 8]
```diff
@@ -368,14 +371,12 @@ public void testProvisionPersisError() {
...
-        when(resStorageProvider.registerAndUpdateClusterSpec(any())).thenReturn(CompletableFuture.supplyAsync(() -> {
-            throw err;
-        }));
+        IOException err = new IOException("persist error");
+        when(storageProvider.registerAndUpdateClusterSpec(any())).thenThrow(err);
         when(resProvider.getResponseHandler()).thenReturn(responseHandler);
 
         ActorRef resourceClusterActor = system.actorOf(
-                ResourceClustersHostManagerActor.props(resProvider, resStorageProvider));
+                ResourceClustersHostManagerActor.props(resProvider, new FileBasedPersistenceProvider(false)));
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/test/java/io/mantisrx/master/resourcecluster/ResourceClustersHostManagerActorTests.java:379]: replace this with something else.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/store/KeyValueStore.java [hunk 9]
```diff
@@ -214,4 +224,61 @@ public boolean deleteAll(String tableName, String partitionKey) {
...
+            }
+            return false;
+        }
+
+        @Override
+        public boolean deleteAll(String tableName, String partitionKey) throws IOException {
+            if (store.containsKey(tableName) && // table exists
+                store.get(tableName).containsKey(partitionKey)) { // partitionKey exists
+                store.get(tableName).remove(partitionKey);
+                return true;
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/store/KeyValueStore.java:279]: @hmit Can you check if the semantics of this is consistent with the KeyValueStore assumptions

**Review by hmitnflx — APPROVED** [review]
looks good, thanks for this change!

**Pull request merged**